File: biopython_rewrite/sam_stats.py
#!/usr/bin/env python3
# Basic SAM stats
import sys
if len(sys.argv)<2:
    print('Usage: sam_stats.py file.sam'); sys.exit(1)
mapped=0; total=0
with open(sys.argv[1]) as fh:
    for line in fh:
        if line.startswith('@'): continue
        total+=1
        cols=line.split('\t')
        if cols[2] != '*': mapped+=1
print(f"Mapped: {mapped} / {total}")

File: legacy_perl/sam_stats.pl
#!/usr/bin/env perl
# Basic SAM flag/statistics parser
use strict;
use warnings;
my $sam = shift or die "Usage: $0 file.sam\n";
open my $fh, '<', $sam or die $!;
my ($mapped,$total)=(0,0);
while(<$fh>){
    next if /^@/;
    $total++;
    my @f = split /\t/;
    $mapped++ if $f[2] ne '*';
}
print "Mapped: $mapped / $total\n";
